src/main/java/com/openshift/internal/restclient/model/ServiceAccount.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package com.openshift.internal.restclient.model;
+
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.model.serviceaccount.IServiceAccount;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * @author David Simansky | dsimansk@redhat.com
+ */
+public class ServiceAccount extends KubernetesResource implements IServiceAccount {
+
+	public ServiceAccount(ModelNode node, IClient client, Map<String, String[]> propertyKeys) {
+		super(node, client, propertyKeys);
+	}
+
+	@Override
+	public Collection<String> getSecrets() {
+		Collection<String> secrets = new ArrayList<>();
+		ModelNode node = get(SERVICE_ACCOUNT_SECRETS);
+		if(node.getType() != ModelType.LIST) return secrets;
+		for (ModelNode entry : node.asList()) {
+			secrets.add(entry.get("name").asString());
+		}
+		return secrets;
+	}
+
+	@Override
+	public void addSecret(String secret) {
+		ModelNode secretNode = get(SERVICE_ACCOUNT_SECRETS);
+		secretNode.add("name", secret);
```

Comment:
Hi @jcantrill,
I've check `JBossDmrExtensions` and `KubernetesResource#set()` methods. Every variant uses combination of `get()` and `set()` on ModelNode object to retriev entry and change value, but I'd ideally need `add()` method to new pair of secret `name: value` Is it sufficient to add null checks or do you have any other suggestion how to proceed?